perceval/simulators/simulator.py

```diff
@@ -65,6 +65,9 @@ def precision(self, value: float):
         assert isinstance(value, Number) and value >= 0., "Precision must be a positive number"
```

Comment:
raise ValueError ?

For above, L63, is @{property_name}.setter is a classical pythonique stuff ?

Reply:
Seems like a good use of assert to me.

Yes `@<property>.setter` is standard Python:
https://realpython.com/python-getter-setter/